### PR TITLE
Add telnet support

### PR DESCRIPTION
**Pull request by ktbyers**
<!-- Make sure you have read http://napalm.readthedocs.io/en/latest/contributing/index.html --!>

### CODE REVIEW

**Comment by ktbyers** [comment]
This expands on earlier telnet support PRs mainly fixing `is_alive` to work with telnet.

I tested telnet connectivity to real device for `get_facts`. I also tested `is_alive()` to real device for both SSH and telnet (in both the alive and fail case).

@mirceaulinic I think we should make the is_alive code referenced here the standard for Netmiko (the SSH section of it). Basically, I used a lower level `write_channel()` for the code which makes more sense given what we are doing.

I also removed some legacy code that checked Netmiko version, but is no longer relevant (since requirements.txt requires a much higher version of Netmiko than what was being checked for).

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -166,21 +171,30 @@ def _send_command(self, command):
...
+        """Returns a flag with the state of the connection."""
         null = chr(0)
-        try:
-            if self.device is None:
+        if self.device is None:
+            return {'is_alive': False}
+        if self.transport == 'telnet':
+            try:
+                # Try sending IAC + NOP
+                self.device.write_channel(telnetlib.IAC + telnetlib.NOP)
```
Comment (mirceaulinic) [inline napalm_ios/ios.py]: That's interesting... Maybe worth adding a comment to explain (I had to look into to RFC to understand this).

**Review by mirceaulinic — APPROVED** [review]
Looks great!

**Comment by ktbyers** [comment]
Okay, expanded comment a bit on IAC + NOP meaning.

**Comment by dbarrosop** [comment]
+1

**Pull request merged**